src/server/common/helpers/proxy/proxy.js

```diff
@@ -0,0 +1,25 @@
+import { ProxyAgent, setGlobalDispatcher } from 'undici'
+import { bootstrap } from 'global-agent'
+import { createLogger } from '~/src/server/common/helpers/logging/logger.js'
+import { config } from '~/src/config/config.js'
+const logger = createLogger()
+
+/**
+ * If HTTP_PROXY is set setupProxy() will enable it globally
+ * for a number of http clients.
+ * Node Fetch will still need to pass a ProxyAgent in on each call.
```

Comment:
So currently we use `node-fetch` in this repo. But the aim is to remove it and use native `fetch` as we are now on Node 22 and this comes without an experimental flag. 

Maybe when we remove `node-fetch` this comment can be updated. I wonder does the `setGlobalDispatcher` work with native Node.js `fetch` or will we have to do something like - https://github.com/DEFRA/cdp-user-service-backend/blob/main/src/helpers/proxy.js#L72-L75

Reply:
From the previous spike `setGlobalDispatcher` only works in unidici. Native fetch requires it to be set on a per-request basis. 
I'd imagine if we wanted to use native fetch we'd just add the proxy into the existing fetcher helper (where we're doing the tracing). Forward-proxy means we don't need to decide at an app level what's proxied and whats not, we can just push everything via it and let forward-proxy work it out.